PWG/EMCAL/EMCALtasks/AliEmcalCorrectionClusterEnergyShift.cxx

```diff
@@ -0,0 +1,104 @@
+// AliEmcalCorrectionClusterEnergyShift
+//
+
+#include "AliEmcalCorrectionClusterEnergyShift.h"
+
+#include "AliClusterContainer.h"
+
+/// \cond CLASSIMP
+ClassImp(AliEmcalCorrectionClusterEnergyShift);
+/// \endcond
+
+// Actually registers the class with the base class
+RegisterCorrectionComponent<AliEmcalCorrectionClusterEnergyShift> AliEmcalCorrectionClusterEnergyShift::reg("AliEmcalCorrectionClusterEnergyShift");
+
+/**
+ * Default constructor
+ */
+AliEmcalCorrectionClusterEnergyShift::AliEmcalCorrectionClusterEnergyShift() :
+  AliEmcalCorrectionComponent("AliEmcalCorrectionClusterEnergyShift"),
+  fEnergyScaleShift(0.),
+  fSmearingWidth(0.),
+  fRandom(0)
+{
+}
+
+/**
+ * Destructor
+ */
+AliEmcalCorrectionClusterEnergyShift::~AliEmcalCorrectionClusterEnergyShift()
+{
+}
+
+/**
+ * Initialize and configure the component.
+ */
+Bool_t AliEmcalCorrectionClusterEnergyShift::Initialize()
+{
+  // Initialization
+  AliEmcalCorrectionComponent::Initialize();
+  
+  GetProperty("energyScaleShift", fEnergyScaleShift);
+  GetProperty("smearingWidth", fSmearingWidth);
+
+  return kTRUE;
+}
+
+/**
+ * Create run-independent objects for output. Called before running over events.
+ */
+void AliEmcalCorrectionClusterEnergyShift::UserCreateOutputObjects()
+{   
+  AliEmcalCorrectionComponent::UserCreateOutputObjects();
+}
+
+/**
+ * Called before the first event to initialize the correction.
+ */
+void AliEmcalCorrectionClusterEnergyShift::ExecOnce()
+{
+  fRandom.SetSeed(0);
+}
+
+/**
+ * Called for each event to process the event data.
+ */
+Bool_t AliEmcalCorrectionClusterEnergyShift::Run()
+{
+  AliEmcalCorrectionComponent::Run();
+  
+  // Loop over all clusters
+  AliTLorentzVector clusFourVec;
+  AliVCluster *cluster = 0;
+  AliClusterContainer * clusCont = 0;
+  TIter nextClusCont(&fClusterCollArray);
+  while ((clusCont = static_cast<AliClusterContainer*>(nextClusCont()))) {
+    auto clusItCont = clusCont->all_momentum();
+    for (AliClusterIterableMomentumContainer::iterator clusIterator = clusItCont.begin(); clusIterator != clusItCont.end(); ++clusIterator) {
+      
+      // Get the cluster energy (using the default energy type set in the cluster container)
+      clusFourVec.Clear();
+      clusFourVec = clusIterator->first;
+      Double_t energyclus = clusFourVec.E();
+
+      // Scale energy (if applicable)
+      if (TMath::Abs(fEnergyScaleShift) > 1e-6) {
+        energyclus *= (1 + fEnergyScaleShift);
+      }
+      
+      // Smear energy (if applicable)
+      if (fSmearingWidth > 1e-6) {
+        energyclus *= (1 + fRandom.Gaus(0, fSmearingWidth) );
+      }
+      
+      // Set the shifted energy to the user-defined energy field
+      cluster = static_cast<AliVCluster *>(clusIterator->second);
+      if (cluster->IsEMCAL() && energyclus > 0) {
+        cluster->SetUserDefEnergy(AliVCluster::kUserDefEnergy1, energyclus);
```

Comment:
Per our conversation, I think this is a scaling factor

Reply:
Never mind, this is fine. In [`AliVCluster`](https://github.com/alisw/AliRoot/blob/f7eab9024d7187361500944ca662e0f1c9c57333/STEER/AOD/AliAODCaloCluster.h#L136), it is set as:

```c++
  void        SetUserDefEnergy(VCluUserDefEnergy_t t, Double_t e)               { fUserDefEnergy[t] = E() > 1e-6 ? e / E() : 1.  ; }
```